src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -12,14 +13,25 @@ namespace System.Text.Json.Serialization
 {
     internal abstract class JsonPropertyInfo
     {
-        // Cache the array and enumerable converters so they don't get created for every enumerable property.
+        // Cache the converters so they don't get created for every enumerable property.
         private static readonly JsonEnumerableConverter s_jsonArrayConverter = new DefaultArrayConverter();
         private static readonly JsonEnumerableConverter s_jsonEnumerableConverter = new DefaultEnumerableConverter();
 
         internal ClassType ClassType;
 
-        internal byte[] _name = default;
-        internal byte[] _escapedName = default;
+        // The name of the property with any casing policy or the name specified from JsonPropertyNameAttribute.
+        private byte[] _name { get; set; }
+        internal ReadOnlySpan<byte> Name => _name;
+        internal string NameAsString { get; private set; }
+
+        // Used to support case-insensitive comparison
+        private byte[] _compareName { get; set; }
+        internal ReadOnlySpan<byte> CompareName => _compareName;
```

Comment:
Are these really necessary? If so, maybe `CompareNameAsSpan`?

Reply:
It avoids casts in other locations.